src/renderer/entities/transaction/lib/transactionService.ts

```diff
@@ -363,6 +369,76 @@ function verifySignature(payload: Uint8Array, signature: HexString, accountId: A
   }
 }
 
+function formatArg(arg: Codec, chain: Chain): unknown {
+  if (Array.isArray(arg)) {
+    return arg.map((a) => formatArg(a, chain));
+  }
+
+  if (arg instanceof GenericCall) {
+    const args: Record<string, unknown> = {};
+
+    for (const [key, value] of arg.argsEntries) {
+      args[key] = formatArg(value, chain);
+    }
+
+    return {
+      section: arg.section,
+      method: arg.method,
+      args,
+    };
+  }
+
+  if (arg instanceof GenericMultiAddress) {
+    if (arg.type === 'Id' || arg.type === 'Address20' || arg.type === 'Address32') {
+      return toAddress(arg.value.toString(), { prefix: chain.addressPrefix });
+    } else {
+      return arg.toHuman();
+    }
+  }
+
+  const isAccount = pjsSchema.accountId.safeParse(arg);
```

Comment:
Not any 32/20 byte arrays are account ids. For example, there are plenty of 32-byte hashes, e.g. call hash, block hash, e.t.c

Instead, we should check whether `arg` is an instance of `AccountId32`/`AccountId20`